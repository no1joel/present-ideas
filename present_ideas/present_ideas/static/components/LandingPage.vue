<template>
  <Layout>
    <div class="container-fluid">
      <div class="row">
        <div class="col">
          <div class="jumbotron">
            <p>Please pick <strong>your name</strong> to get started.</p>
            <div class="form-row">
              <PeoplePicker
                class="col-sm-10"
                v-bind:loading="loading"
                v-bind:people="people"
                v-model="currentUser"
              />
              <button
                type="button"
                class="btn btn-block btn-success col-sm-2"
                v-bind:disabled="!currentUser"
              >
                Go
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Layout>
</template>

<script>
import Layout from "../layout/Layout.vue";
import PeoplePicker from "./PeoplePicker.vue";

export default {
  components: {
    Layout,
    PeoplePicker,
  },
  data() {
    return {
      people: [],
      loading: true,
      currentUser: undefined,
    };
  },
  mounted() {
    this.loadPeople();
  },
  methods: {
    async loadPeople() {
      this.loading = true;
      const response = await fetch("/api/person");
      const data = await response.json();
      const names = data.names;
      this.people = names.sort();
      this.loading = false;
    },
  },
};
</script>