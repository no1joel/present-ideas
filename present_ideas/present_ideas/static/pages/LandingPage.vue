<template>
  <Layout>
    <div class="container-fluid">
      <div class="row">
        <div class="col">
          <div class="jumbotron">
            <p>Please pick <strong>your name</strong> to get started.</p>
            <div class="form-row">
              <PeoplePicker
                class="col-sm-10"
                v-bind:loading="loading"
                v-bind:people="people"
                v-model="currentUser"
              />
              <button
                type="button"
                class="btn btn-block btn-success col-sm-2"
                v-bind:disabled="!currentUser"
                v-on:click="goToMyPage"
              >
                Go
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Layout>
</template>

<script>
import Layout from "../layout/Layout.vue";
import PeoplePicker from "../components/PeoplePicker.vue";

export default {
  components: {
    Layout,
    PeoplePicker,
  },
  data() {
    return {
      people: [],
      loading: true,
      currentUser: undefined,
    };
  },
  mounted() {
    this.loadPeople();
  },
  methods: {
    async loadPeople() {
      this.loading = true;
      const response = await fetch("/api/person");
      const data = await response.json();
      const names = data.names;
      this.people = names.sort();
      this.loading = false;
    },
    goToMyPage() {
      this.$router.push({ name: "myList", params: { user: this.currentUser } });
    },
  },
};
</script>