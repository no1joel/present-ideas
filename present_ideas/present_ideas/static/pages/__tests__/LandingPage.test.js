import { render } from "@testing-library/vue";
import { createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import getStore from "../../store";
import LandingPage from "../LandingPage.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("LandingPage", () => {
  it("matches snapshot", () => {
    const store = getStore();
    expect(
      render(LandingPage, { localVue, store }).container
    ).toMatchSnapshot();
  });
});
