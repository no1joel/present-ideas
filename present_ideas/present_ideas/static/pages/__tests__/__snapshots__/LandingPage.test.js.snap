// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage matches snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-light bg-light"
    >
      <router-link
        class="navbar-brand"
        to="/"
      >
        Present Ideas
      </router-link>
       
      <!---->
    </nav>
     
    <div
      class="mt-3"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="jumbotron"
            >
              <p>
                Pick 
                <strong>
                  your name
                </strong>
                 to get started ðŸ˜Š
              </p>
               
              <div
                class="form-row"
              >
                <select
                  class="custom-select col-sm-10"
                >
                  <!---->
                   
                  <option
                    disabled="disabled"
                    value=""
                  >
                    
    Pick a name!
  
                  </option>
                   
                </select>
                 
                <button
                  class="btn btn-block btn-success col-sm-2"
                  disabled="disabled"
                  type="button"
                >
                  
              Go
            
                </button>
              </div>
               
              <p>
                <small
                  class="text-muted"
                >
                  
              I need to know who you are, first, so that when you're looking
              at your own list you can't see who's getting you what -- no
              surprises! I've not set up proper usernames and passwords
              because that's a faff. If you don't see your name there,
              seriously who are you and how did you get here?
            
                </small>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
